include/bluetooth/gatt.h

```diff
@@ -1208,10 +1208,30 @@ struct bt_gatt_exchange_params {
  *
  *  @note Shall only be used once per connection.
  *
+ *  The Response comes in callback @p params->func. The callback is run from
+ *  the Bluetooth event context. @p *params must remain valid until start of
+ *  callback.
+ *
+ *  This function will block while the ATT request queue is full, except when
+ *  called from Bluetooth event context. When called from Bluetooth context,
+ *  this function will instead instead return `-ENOMEM` if it would block to
+ *  avoid a deadlock.
+ *
  *  @param conn Connection object.
  *  @param params Exchange MTU parameters.
  *
  *  @return 0 in case of success or negative value in case of error.
+ *
+ *  @retval 0
+ *      Successfully queued request.
+ *      Will call @p params->func on resolution.
+ *
+ *  @retval -ENOMEM
+ *      ATT request queue is full and blocking would cause deadlock.
+ *      Allow a pending request to resolve before retrying, or call
+ *      this function outside Bluetooth event context to get blocking
+ *      behavior.
+ *      Queue size is controlled by @kconfig{CONFIG_BT_L2CAP_TX_BUF_COUNT}.
```

Comment:
Uh, I like the use of `@retval`. That's something I'd like to see more of in the Bluetooth headers